src/data/list/nodup.lean

```diff
@@ -18,7 +18,7 @@ universes u v
 
 open nat function
 
-variables {α : Type u} {β : Type v} {l l₁ l₂ : list α} {a b : α}
+variables {α : Type u} {β : Type v} {l l₁ l₂ : list α} {r : α → α → Prop} {a b : α}
```

Comment:
Does this result in better argument order? Seems no because `nodup.pairwise_coe` would have `{r : α → α → Prop} {a : α} [is_symm α r]` but it's better that `[is_symm α r]` immediately follows `{r}`?

Reply:
Surely it's better that `r` be before `a` and `b`? `r` is a relation, and `a`, `b` are single elements.

Reply:
But I don't understand why you change it at this `variables` declaration. You didn't change existing def/lemmas so this change only applies to `nodup.pairwise_coe`, but you may as well do
```
@[simp] lemma nodup.pairwise_coe {r : α → α → Prop} [is_symm α r] {a : α} (hl : l.nodup) :
```

Reply:
Oh yes, what you've done works. I was confused; the variable `a : α` isn't used in `nodup.pairwise_coe`.